next/content/algorithms/sdr.md

```diff
@@ -0,0 +1,1897 @@
+* [`storage_proofs::core::merkle::proof::InclusionPath::root()`](https://github.com/filecoin-project/rust-fil-proofs/blob/f75804c503d9b97a2b02ef3ea2e5d44e8e2c6470/storage-proofs/core/src/merkle/proof.rs#L189)
+
+```text
+$\overline{\underline{\Function \bintreeproofisvalid(\proof: \BinTreeProof) \rightarrow \Bool}\thin}$
+$\line{1}{\bi}{\curr: \Safe = \proof\dot\leaf}$
+$\line{2}{\bi}{\for \sibling, \missing \in \proof\dot\path:}$
+$\line{3}{\bi}{\quad \curr: \Safe = \if \missing = 0:}$
+$\quad\quad\quad\quad \Sha{254}_2([\curr, \sibling])$
+$\quad\quad\thin \else:$
+$\quad\quad\quad\quad \Sha{254}_2([\sibling, \curr])$
+$\line{4}{\bi}{\return \curr = \proof\dot\root}$
+```
+
+The function `$\octtreeproofisvalid$` can receive as the type of its `$\proof$` argument either an `$\OctTreeProof$` or `$\ColumnProof$` (a `$\ColumnProof$` is just an `$\OctTreeProof$` with an adjoined field `$\column$`, `$\ColumnProof_c \equiv \OctTreeProof_c \cup \column_c \thin$`).
+
+```text
+$\overline{\underline{\Function \octtreeproofisvalid(\proof: \OctTreeProof) \rightarrow \Bool}\thin}$
+$\line{1}{\bi}{\curr: \Fq = \proof\dot\leaf}$
+$\line{2}{\bi}{\for \siblings, \missing \in \proof\dot\path:}$
+$\line{3}{\bi}{\quad \inputs: \Fq^{[8]} = \siblings[\ldotdot \missing] \concat \curr \concat \siblings[\missing \ldotdot]}$
+$\line{4}{\bi}{\quad \curr = \Poseidon_8(\inputs)}$
+$\line{5}{\bi}{\return \curr = \proof\dot\root}$
+```
+
+```text
+$\overline{\underline{\Function \octtreeproofisvalid(\proof: \ColumnProof) \rightarrow \Bool}\thin}$
+$\line{1}{\bi}{\return \octtreeproofisvalid(\OctTreeProof\ \{\ \leaf, \root, \path \Leftarrow \ColumnProof\ \})}$
+```
+
+### Merkle Proof Challenge Validation
+
+Given a Merkle path `$\path$` in a `$\BinTree$` or `$\OctTree$`, `$\calculatebintreechallenge$` and `$\calculateocttreechallenge$` calculate the Merkle challenge `$c$` for which the Merkle proof `$\path$` was generated.
+
+Given a Merkle challenge `$c$` and its path in a `$\BinTree$` or `$\OctTree$`, the concatentation of the `$\missing$` bits (or octal digits) in the Merkle path is the little-endian binary (or octal) representation of the integer `$c \thin$`:
+
+```text
+$\line{1}{\bi}{c: \NodeIndex = \langle \text{some challenge} \rangle}$
+
+$\line{2}{\bi}{\BinTreeProof_c = \BinTree\dot\createproof(c)}$
+
+$\line{3}{\bi}{\OctTreeProof_c = \OctTree\dot\createproof(c)}$
+
+$\line{4}{\bi}{\llcorner c \lrcorner_{2, \textsf{le}}: \Bit^{[\log_2(N_\nodes)]} = \Big\|_{\el \in \BinTreeProof_c\dot\path} \thin \el\dot\missing}$
+
+$\line{5}{\bi}{\llcorner c \lrcorner_{8, \textsf{le}}: [8]^{[\log_8(N_\nodes)]} \ \ \ \ \ \thin = \Big\|_{\el \in \OctTreeProof_c\dot\path} \thin \el\dot\missing}$
+
+$\line{6}{\bi}{\llcorner c \lrcorner_{2, \textsf{le}}: \Bit^{[\log_2(N_\nodes)]} = \Big\|_{\el \in \OctTreeProof_c\dot\path} \thin \llcorner \el\dot\missing \lrcorner_{2, \textsf{le}}}$
```

Comment:
I'd leave that to you :) Sometimes in the source you're using `\|` directly and sometimes the own defined `\concat`. I don't have the context on whether in makes sense to always use `\concat`. If it would be the case, it would be really easy to change it to `\Big\|_{}`.

Reply:
Those fixes also went in. Please note that sadly the rendering of `\big` looks quite ugly in KaTeX (at least on my machine).